sktime/networks/ltsf/data/dataset.py

```diff
@@ -0,0 +1,77 @@
+"""Implements Pytorch Dataset class for LTSF-Formers."""
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+from sktime.networks.ltsf.utils.timefeatures import time_features
+
+
+class PytorchFormerDataset(Dataset):
+    """Implements Pytorch Dataset class for LTSF-Formers."""
+
+    def __init__(
+        self,
+        y,
+        seq_len,
+        context_len,
+        pred_len,
+        freq,
+        timeenc,  # boolean
+    ):
+        self.y = y
+        self.seq_len = seq_len
+        self.context_len = context_len
+        self.pred_len = pred_len
+        self.freq = freq
+        self.timeenc = timeenc
+
+        self._prepare_data()
+
+    def _prepare_data(self):
```

Comment:
summarizing discussion from meeting with @benHeid:

* the `_prepare_data` method seems extraneous to the neural network and duplicative with `DateTimeFeatures` transformation.
* so, we should check whether it could be replaced with `DateTimeFeatures` in a pipeline.
* For this, the user needs to be able to pass arbitrary exogeneous data. We should check if that is easily possible.
* optional: for retaining reference behaviour, we could give the user a flag to switch `_prepare_data` on or off, but we thought that it might complicate things too much, so removal and replacement by an exogeneous input might be preferable.
    * overal, for the options remove/switch etc, there is no strict "must", but we should do what is easiest and/or leads to the simplest design (as opposed to overcomplication and obscure interface)

Reply:
Here are some more insights that I already shared in the discord channel

---

The current architechture uses these 3 embeddings

1. value_embedding: uses the values in y
2. position_embedding: uses the relative indexes in y (0, 1, 2, ...)
3. temporal_embedding: uses the frequency and datetime index

We can completely remove temporal_embedding (thats what Nixtla is doing) because
- it involves a lot of complexities
- hard coding
- requirements with the base class
- depends on the frequency in data index
- not adding much value to the model predictions

---

But if we want to keep the temporal_embedding, we can do that by refactoring the existing code to better protect from future breaks (but the vulnerability exists)

- Model architechture depends on the shape of input data and the range of values in input data
- the shape and range of input data depends on
    - frequency of index
    - transformation method applied
    - presence of exogenous data

we can deduce how the input data will look like from the parameters

---

exog data is another story, we cannot replace it with temporal_embedding (they are conceptually different). Although we can add support for exog data along with the existing embeddings as a 4th embeddings - but again that is another discussion.

---